[conf] Gracefully handle the change of config options for modifying playlists

chme opened the pull request:
@ejurgensen I think it would be best not to break existing config files. This pr readds the old options in the 'mpd' section and displays a deprecation error in the log if they are present.

In a future version the deprecated options could then be dropped completely.

ejurgensen commented:
Yes, agree, what do you think about using CFGF_DEPRECATED and/or CFGF_DROP? See https://github.com/martinh/libconfuse/issues/24

Then perhaps the "Handle deprecated config options" would not be required. In any case I think that code should be in mpd.c, not in conffile.c.

BTW maybe "default_playlist_directory" should be called "saved_playlists_dir": 1) we use "dir" later in the config file, but "directory" is not used in other option names, 2) I don't really understand the "default" part...

chme commented:
I've updated the pr to use CFGF_DEPRECATED and moved the deprecation handling to mpd.c.
CFGF_DROP would only be useful, if we would write the config file after parsing it.

In order to get the deprecation messages from libconfuse in the forked-daapd log, it is necessary to set `cfg_set_error_function`. I don't really like the modification i have done to logger.c/h, but could not think of a better approach.

There seems to be a bug in libconfuse that leads to multiple log entries for a deprecated option (i have to investigate further before opening an issue for libconfuse - at first glance this seems to happen, if the deprecated option is followed by comments). But having multiple entries in the log is better than none :-)

ejurgensen commented:
Merged, thanks!
